backend/src/main/java/com/empire/EntryServlet.java

```diff
@@ -204,6 +225,21 @@ private String getWorld(Request r) {
 		}
 	}
 
+	private String getGeography(Request r) {
+		try {
+			DatastoreService service = DatastoreServiceFactory.getDatastoreService();
+			int date = r.turn != 0 ? r.turn : getWorldDate(r.gameId, service);
```

Comment:
Thanks for pointing this undocumented confusion out. I've created .hasTurn in request which will help clear it up a little bit, although it could be further improved. The idea is:
 • A world never has the date as 0. (The first turn is date == 1).
 • If a request has turn = 0, it either didn't have the "t" query parameter or t=0. Either way, it either doesn't care about the turn (e.g. "setup, startNew*" fall into this case) or it means the latest turn (e.g. the initial fetch from the front-end when a player logs in is asking for the latest turn of data).

Request in general was a very quick hack and has gotten pretty ugly in how it is used. I think it'd be nicer to have a Request/Response class per action the servlet accepts that more clearly call out the expected data; but postponing that to another PR.